frontend/src/app/Settings/Users/ManageUsers.tsx

```diff
@@ -120,35 +115,103 @@ const ManageUsers: React.FC<Props> = ({
     updateIsUserEdited(false);
   };
 
-  const onHandleCreateNewUser = (newUserInvite: NewUserInvite) => {
-    // TODO: validate form
-    onCreateNewUser(newUserInvite);
+  const handleUpdateUser = (userId: string) => {
+    const selectedRoleDescription = usersState[userId]
+      .roleDescription as RoleDescription;
+    const selectedOrganizationRole = RoleDescriptionToOrgRole[
+      selectedRoleDescription
+    ] as OrganizationRole;
+    updateUserRole({
+      variables: {
+        id: userId,
+        role: selectedOrganizationRole,
+      },
+    })
+      .then(() => {
+        updateIsUserEdited(false);
 
-    const fullName = displayFullName(
-      newUserInvite.firstName,
-      null,
-      newUserInvite.lastName
-    );
+        const user = usersState[userId];
 
-    let successAlert = (
-      <Alert
-        type="success"
-        title={`Invitation sent to ${fullName}`}
-        body={`They will receive an invitation to create an account at the email address provided`}
-      />
-    );
+        const fullName = displayFullNameInOrder(
+          user.firstName,
+          user.middleName,
+          user.lastName
+        );
 
-    showNotification(toast, successAlert);
-    updateShowAddUserModal(false);
+        showNotification(
+          toast,
+          <Alert
+            type="success"
+            title="Changes Saved"
+            body={`${fullName}'s settings have been saved`}
+          />
+        );
+      })
+      .catch(setError);
   };
 
-  const onHandleDeleteUser = (userId: string) => {
-    // TODO: show alert only if delete is successful
-    onDeleteUser(userId);
+  const handleAddUserToOrg = (newUserInvite: NewUserInvite) => {
+    // TODO: validate form
+    const { firstName, lastName, email } = { ...newUserInvite };
+    addUserToOrg({
+      variables: {
+        firstName: firstName,
+        lastName: lastName,
+        email: email,
+      },
+    })
+      .then(() => {
+        const fullName = displayFullNameInOrder(firstName, "", lastName);
+
+        showNotification(
+          toast,
+          <Alert
+            type="success"
+            title={`Invitation sent to ${fullName}`}
+            body={`They will receive an invitation to create an account at the email address provided`}
+          />
+        );
+        updateShowAddUserModal(false);
+        getUsers();
```

Comment:
Still trying to understand the exact semantics of `useLazyQuery`. I gather this actually fires off the query, and sometime later you get an updated list? Is there any negative aspect of having a stale list for a short time?

Reply:
We actually don't end up with a stale list because in the Container component we show a loading indicator while the user query is executing. So for better or worse whenever we call `getUsers()` we end up seeing a loading indicator.